exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/lateraljoin/TestE2EUnnestAndLateral.java

```diff
@@ -114,6 +115,57 @@ public void testMultiUnnestAtSameLevel() throws Exception {
     test(Sql);
   }
 
+  @Test
+  public void testUnnestWithItem() throws Exception {
+    String sql = "select u.item from\n" +
+        "cp.`lateraljoin/nested-customer.parquet` c," +
+        "unnest(c.orders['items']) as u(item)\n" +
+        "limit 1";
+
+    testBuilder()
+        .sqlQuery(sql)
+        .unOrdered()
+        .baselineColumns("item")
+        .baselineValues(
+            TestBuilder.mapOf("i_name", "paper towel",
+                "i_number", 2.0,
+                "i_supplier", "oregan"))
+        .go();
+  }
+
+  @Test
+  public void testUnnestWithFunctionCall() throws Exception {
+    String sql = "select u.ord.o_amount o_amount from\n" +
+        "cp.`lateraljoin/nested-customer.parquet` c," +
+        "unnest(convert_fromjson(convert_tojson(c.orders))) as u(ord)\n" +
+        "limit 1";
+
+    testBuilder()
+        .sqlQuery(sql)
+        .unOrdered()
+        .baselineColumns("o_amount")
+        .baselineValues(4.5)
+        .go();
+  }
+
+  @Test
+  public void testUnnestWithMap() throws Exception {
+    String sql = "select u.item from\n" +
```

Comment:
Can you also add multi unnest query to the testing. (something like this)
select u.item from cp.`lateraljoin/nested-customer.parquet` c, unnest(c.orders.items) as u(item), unnest(c.orders.items) as u1(item1) limit 1

Reply:
Thanks, added `testMultiUnnestWithMap()` test.